pkg/autoupdate/findnew.go

```diff
@@ -141,28 +142,24 @@ func FindNewest(ctx context.Context, fullBinaryPath string, opts ...newestOption
 		"binaryName", binaryName,
 	)
 
-	// Find the possible updates. filepath.Glob returns a list of things
-	// that match the requested pattern. We sort the list to ensure that
-	// we can tell which ones are earlier or later (remember, these are
-	// timestamps). If no updates are found, the forloop is skipped, and
-	// we return eithier the seed fullBinaryPath or ""
-	fileGlob := filepath.Join(updateDir, "*", binaryName)
-
-	possibleUpdates, err := filepath.Glob(fileGlob)
+	// If no updates are found, the forloop is skipped, and we return either the seed fullBinaryPath or ""
+	possibleUpdates, err := getPossibleUpdates(updateDir, binaryName)
 	if err != nil {
-		level.Error(logger).Log("msg", "globbing for updates", "err", err)
+		level.Error(logger).Log("msg", "could not find possible updates", "err", err)
 		return fullBinaryPath
 	}
 
-	sort.Strings(possibleUpdates)
-
 	// iterate backwards over files, looking for a suitable binary
 	foundCount := 0
 	foundFile := ""
 	for i := len(possibleUpdates) - 1; i >= 0; i-- {
 		file := possibleUpdates[i]
 		basedir := filepath.Dir(file)
 		updateDownloadTime := filepath.Base(basedir)
+		foundExecutable := file
+		if strings.HasSuffix(file, ".app") {
+			foundExecutable = filepath.Join(file, "Contents", "MacOS", binaryName)
+		}
```

Comment:
See comment below -- we add and remove and add this again

Reply:
Replied [here](https://github.com/kolide/launcher/pull/932#discussion_r1021974705)